src/array.vala

```diff
@@ -2,70 +2,86 @@ using GLib;
 
 public class Vast.Array : Object
 {
-    public Type scalar_type { get; construct; }
+    /* GType of the scalar elements, immutable */
+    public Type scalar_type { get; construct; /* implicitly typeof (void) */ }
```

Comment:
This is a really tricky thing because when I set it explicitly, it crashes. It would probably be a good idea to investigate.